squape/object_tree.py

```diff
@@ -0,0 +1,215 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023, Cyber Alpaca
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import math
+import types
+
+try:
+    import squish
+except ImportError:
+    import squishtest as squish
+
+import object
+
+
+def children(object_name: any, selector: dict = None) -> tuple:
+    """Filter the direct children of the specified object by the specified selector
+
+    Args:
+        object_name (any): symbolic name, real name, or object reference to the parent
+        selector (dict, optional): The selector is a dictionary of properties,
+        that must match for objects to be included into the result. 
+        The values of the selector can be lambda functions as well.
+        In that case, the verification would consist of passing the candidate's key attribute into that function. 
+        The return value should always be bool.
+        Defaults to {}, which means all objects pass the verification.
+
+    Returns:
+        tuple: children objects that met the selector criteria
+
+    Examples:
+        >>> children(object_name, {'type' : Button, 'visible' : True})
+    """
+    if selector is None:
+        selector = {}
+    object_reference = _get_object_reference(object_name)
+    children = object.children(object_reference)
+    return tuple(filter(lambda x: _is_matching(x, selector), children))
+
+
+def find(object_name: any, selector: dict = None, max_depth: int = None) -> tuple:
+    """Recursively filter all the children of the specified object
```

Comment:
```suggestion
    """Finds all the children of the given object.
```

Reply:
"Finds all descendants of the given object. "